### PR TITLE
Add support for AoP.

### PR DESCRIPTION
**Pull request by MarvinCai**

### CODE REVIEW

File: charts/sn-platform/values.yaml [hunk 1]
```diff
@@ -934,6 +936,9 @@ broker:
   advertisedPodIP: false
   # For exposing kop service for external access
   advertisedDomain: ""
+  dnsNames:
```
Comment (fantapsody) [inline charts/sn-platform/values.yaml:939]: Would it be better to make it an empty list by default and put those DNS name examples in comments?

File: charts/sn-platform/templates/broker/broker-cluster.yaml [hunk 2]
```diff
@@ -183,6 +187,13 @@ spec:
             key: password
     {{- end }}
     {{- end }}
+    {{- if or .Values.broker.aop.enabled .Values.components.aop }}
+      aop:
+        enabled: true
+        {{- if or .Values.broker.aop.proxyEnabled }}
```
Comment (fantapsody) [inline charts/sn-platform/templates/broker/broker-cluster.yaml:193]: Is this statement useful?

File: charts/sn-platform/templates/broker/broker-cluster.yaml [hunk 3]
```diff
@@ -183,6 +187,13 @@ spec:
             key: password
     {{- end }}
     {{- end }}
+    {{- if or .Values.broker.aop.enabled .Values.components.aop }}
```
Comment (fantapsody) [inline charts/sn-platform/templates/broker/broker-cluster.yaml:190]: It's a bit confusing to have multiple flags to enable a feature...

File: charts/sn-platform/values.yaml [hunk 4]
```diff
@@ -945,6 +950,9 @@ broker:
       #  key: password
       #  name: kop-keystore-password
   # choose istio selector
+  aop:
+    enabled: false
+    proxyEnabled: false
```
Comment (fantapsody) [inline charts/sn-platform/values.yaml:955]: Would users prefer the proxy to be enabled by default?

**Review by fantapsody — APPROVED** [review]
Overall looks good, left some comments.

File: charts/sn-platform/values.yaml (hunk 1 above)
Comment (MarvinCai) [reply charts/sn-platform/values.yaml:939]: sure, will comment it out.

File: charts/sn-platform/templates/broker/broker-cluster.yaml (hunk 3 above)
Comment (MarvinCai) [reply charts/sn-platform/templates/broker/broker-cluster.yaml:190]: honestly yes, but I'm not sure why we have 2 different flags, I assume user will expect kop/aop enabled if we set either of them so just check both here.

File: charts/sn-platform/values.yaml (hunk 4 above)
Comment (MarvinCai) [reply charts/sn-platform/values.yaml:955]: I think value's default is for most user who might not need aop enabled, so keep it to false by default.

File: charts/sn-platform/templates/broker/broker-cluster.yaml (hunk 2 above)
Comment (MarvinCai) [reply charts/sn-platform/templates/broker/broker-cluster.yaml:193]: removed the or

**Pull request merged**